### PR TITLE
TuiView

### PR DESCRIPTION
**Pull request by valtandor**
A number of Easybuilds with dependencies for TuiView, a light-weight raster GIS viewer. Requires #3981.

### CODE REVIEW

**Comment by boegel** [comment]
@valtandor two issues: #2834 is blocked, we shouldn't update versions in existing easyconfigs

Also, there's a lot of easyconfigs being added here for X11 libraries; we have a better approach for that now via the `X11` bundles, see https://github.com/hpcugent/easybuild-easyconfigs/tree/master/easybuild/easyconfigs/x/X11.

An easy way out seems to be to update this to a newer toolchain (e.g. `foss/2016b`), and use the readily avaliable `X11` as a dep instead. Is that something worth considering for you?

**Comment by valtandor** [comment]
@boegel I've been working on a complex set of dependencies for a year now, in the context of my site not upgrading toolchains. Moving to a later toolchain for this one package (once they are available) is potentially tedious, but could be considered if there is an easy way to rebuild all the other related packages.

**Comment by boegel** [comment]
@valtandor another alternative is including an easyconfig for the X11 bundle with the toolchain you're using here (which I'd strongly prefer over porting a bunch of X11-related easyconfigs)

**Comment by valtandor** [comment]
@boegel This should be better - I've cleaned out the X11 cruft.

File: easybuild/easyconfigs/q/Qt/Qt-4.8.7-foss-2015a-GLib-2.48.2.eb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+homepage = 'http://qt.io/'
+description = "Qt is a comprehensive cross-platform C++ application framework."
+
+toolchain = {'name': 'foss', 'version': '2015a'}
+
+source_urls = [
+    'http://download.qt.io/official_releases/qt/%(version_major_minor)s/%(version)s/',
+    'http://download.qt.io/archive/qt/%(version_major_minor)s/%(version)s/'
+]
+sources = ['%(namelower)s-everywhere-opensource-src-%(version)s.tar.gz']
```
Comment (verdurin) [inline easybuild/easyconfigs/q/Qt/Qt-4.8.7-foss-2015a-GLib-2.48.2.eb]: @valtandor Please add an `SHA256` checksum here.

File: easybuild/easyconfigs/s/SIP/SIP-4.18.1-foss-2015a-Python-2.7.11.eb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# This file is an EasyBuild reciPY as per https://github.com/hpcugent/easybuild
```
Comment (verdurin) [inline easybuild/easyconfigs/s/SIP/SIP-4.18.1-foss-2015a-Python-2.7.11.eb]: Please fix the URL to use the new `easybuilders` repo address, here and elsewhere.

File: easybuild/easyconfigs/s/SIP/SIP-4.18.1-foss-2015a-Python-2.7.11.eb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+name = 'SIP'
+version = '4.18.1'
+
+homepage = 'http://www.riverbankcomputing.com/software/sip/'
+description = """SIP is a tool that makes it very easy to create Python bindings for C and C++ libraries."""
+
+toolchain = {'name': 'foss', 'version': '2015a'}
+
+sources = [SOURCELOWER_TAR_GZ]
+source_urls=['http://downloads.sourceforge.net/project/pyqt/sip/sip-%(version)s/']
```
Comment (verdurin) [inline easybuild/easyconfigs/s/SIP/SIP-4.18.1-foss-2015a-Python-2.7.11.eb]: Please add a checksum.

File: easybuild/easyconfigs/s/SIP/SIP-4.18.1-foss-2015a-Python-2.7.11.eb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+python = 'Python'
+pyver = '2.7.11'
+versionsuffix = '-%s-%s' % (python, pyver)
+
+dependencies = [(python, pyver)]
+
+pyshortver = '.'.join(pyver.split('.')[:2])
```
Comment (verdurin) [inline easybuild/easyconfigs/s/SIP/SIP-4.18.1-foss-2015a-Python-2.7.11.eb]: Please update this section to use the new built in `pyver` etc. support for Python.

File: easybuild/easyconfigs/s/SIP/SIP-4.18.1-foss-2015a-Python-3.5.1.eb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+name = 'SIP'
+version = '4.18.1'
+
+homepage = 'http://www.riverbankcomputing.com/software/sip/'
+description = """SIP is a tool that makes it very easy to create Python bindings for C and C++ libraries."""
+
+toolchain = {'name': 'foss', 'version': '2015a'}
+
+sources = [SOURCELOWER_TAR_GZ]
+source_urls=['http://downloads.sourceforge.net/project/pyqt/sip/sip-%(version)s/']
```
Comment (verdurin) [inline easybuild/easyconfigs/s/SIP/SIP-4.18.1-foss-2015a-Python-3.5.1.eb]: Please add a checksum.

File: easybuild/easyconfigs/s/SIP/SIP-4.18.1-foss-2015a-Python-3.5.1.eb [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+python = 'Python'
+pyver = '3.5.1'
+versionsuffix = '-%s-%s' % (python, pyver)
```
Comment (verdurin) [inline easybuild/easyconfigs/s/SIP/SIP-4.18.1-foss-2015a-Python-3.5.1.eb]: Please update this to the new style, and move `versionsuffix` to immediately below `version`.

File: easybuild/easyconfigs/s/SIP/SIP-4.19.2-gimkl-2017a-Python-2.7.13.eb [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+description = """SIP is a tool that makes it very easy to create Python bindings for C and C++ libraries."""
+
+toolchain = {'name': 'gimkl', 'version': '2017a'}
+
+sources = [SOURCELOWER_TAR_GZ]
+source_urls = ['http://sourceforge.net/projects/pyqt/files/sip/sip-%(version)s']
+
+dependencies = [('Python', pyver)]
+
+pyshortver = '.'.join(pyver.split('.')[:2])
```
Comment (verdurin) [inline easybuild/easyconfigs/s/SIP/SIP-4.19.2-gimkl-2017a-Python-2.7.13.eb]: Same as above, please update `Python` syntax, and add a `checksum`.

File: easybuild/easyconfigs/s/SIP/SIP-4.19.2-gimkl-2017a-Python-3.6.0.eb [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+description = """SIP is a tool that makes it very easy to create Python bindings for C and C++ libraries."""
+
+toolchain = {'name': 'gimkl', 'version': '2017a'}
+
+sources = [SOURCELOWER_TAR_GZ]
+source_urls = ['http://sourceforge.net/projects/pyqt/files/sip/sip-%(version)s']
+
+dependencies = [('Python', pyver)]
+
+pyshortver = '.'.join(pyver.split('.')[:2])
```
Comment (verdurin) [inline easybuild/easyconfigs/s/SIP/SIP-4.19.2-gimkl-2017a-Python-3.6.0.eb]: Same as above.

File: easybuild/easyconfigs/t/TuiView/TuiView-1.1.11-gimkl-2017a-GLib-2.52.0-GEOS-3.5.1-Python-2.7.13.eb [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+source_urls = ['https://bitbucket.org/chchrsc/tuiview/downloads/']
+sources = [SOURCE_TAR_GZ]
+
+glib_version = '2.52.0'
+geos_version = '3.5.1'
+python_version = '2.7.13'
+glibsuffix = '-GLib-%s' % glib_version
+geossuffix = '-GEOS-%s' % geos_version
+pysuffix = '-Python-%s' % python_version
+versionsuffix = '%s%s%s' % (glibsuffix, geossuffix, pysuffix)
```
Comment (verdurin) [inline easybuild/easyconfigs/t/TuiView/TuiView-1.1.11-gimkl-2017a-GLib-2.52.0-GEOS-3.5.1-Python-2.7.13.eb]: Please use the new `Python` version support and move this section to immediately below `version`.

File: easybuild/easyconfigs/t/TuiView/TuiView-1.1.11-gimkl-2017a-GLib-2.52.0-GEOS-3.5.1-Python-2.7.13.eb [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+version = '1.1.11'
+
+homepage = "https://tuiview.org/"
+description = """TuiView is a lightweight raster GIS with powerful raster attribute table manipulation
+ abilities."""
+
+toolchain = {'name': 'gimkl', 'version': '2017a'}
+
+source_urls = ['https://bitbucket.org/chchrsc/tuiview/downloads/']
+sources = [SOURCE_TAR_GZ]
```
Comment (verdurin) [inline easybuild/easyconfigs/t/TuiView/TuiView-1.1.11-gimkl-2017a-GLib-2.52.0-GEOS-3.5.1-Python-2.7.13.eb]: Please add a checksum.

File: easybuild/easyconfigs/t/TuiView/TuiView-1.1.11-gimkl-2017a-GLib-2.52.0-GEOS-3.5.1-Python-3.6.0.eb [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+source_urls = ['https://bitbucket.org/chchrsc/tuiview/downloads/']
+sources = [SOURCE_TAR_GZ]
+
+glib_version = '2.52.0'
+geos_version = '3.5.1'
+python_version = '3.6.0'
+glibsuffix = '-GLib-%s' % glib_version
+geossuffix = '-GEOS-%s' % geos_version
+pysuffix = '-Python-%s' % python_version
+versionsuffix = '%s%s%s' % (glibsuffix, geossuffix, pysuffix)
```
Comment (verdurin) [inline easybuild/easyconfigs/t/TuiView/TuiView-1.1.11-gimkl-2017a-GLib-2.52.0-GEOS-3.5.1-Python-3.6.0.eb]: Same as above.

File: easybuild/easyconfigs/t/TuiView/TuiView-1.1.9-foss-2015a-GLib-2.48.2-Python-2.7.11.eb [hunk 12]
```diff
@@ -0,0 +1,32 @@
+easyblock = 'PythonPackage'
+
+name = 'TuiView'
+glibver = '2.48.2'
+glibsuffix = '-GLib-%s' % glibver
+pyver = '2.7.11'
+pysuffix = '-Python-%s' % pyver
+version = '1.1.9'
+versionsuffix = '%s%s' % (glibsuffix, pysuffix)
```
Comment (verdurin) [inline easybuild/easyconfigs/t/TuiView/TuiView-1.1.9-foss-2015a-GLib-2.48.2-Python-2.7.11.eb]: Please clean up the `Python` version support.

File: easybuild/easyconfigs/t/TuiView/TuiView-1.1.9-foss-2015a-GLib-2.48.2-Python-2.7.11.eb [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+
+homepage = "https://tuiview.org/"
+description = """TuiView is a lightweight raster GIS with powerful raster attribute table manipulation
+ abilities."""
+
+toolchain = {'name': 'foss', 'version': '2015a'}
+
+source_urls = ['https://bitbucket.org/chchrsc/tuiview/downloads/']
+sources = [SOURCE_TAR_GZ]
```
Comment (verdurin) [inline easybuild/easyconfigs/t/TuiView/TuiView-1.1.9-foss-2015a-GLib-2.48.2-Python-2.7.11.eb]: Please add a checksum.

**Comment by migueldiascosta** [comment]
@valtandor, this PR is being closed for the following reason(s): no activity for > 1 year.
Please don't hesitate to reopen this PR or add a comment if you feel this contribution is still relevant.
For more information on our policy w.r.t. closing PRs, see https://easybuild.readthedocs.io/en/latest/Contributing.html#why-a-pull-request-may-be-closed-by-a-maintainer

**Pull request closed without merging**